libgnucash/engine/gnc-commodity.cpp

```diff
@@ -354,20 +354,20 @@ gnc_quote_source_add_new (const char *source_name, gboolean supported)
     gnc_quote_source *new_source;
 
     DEBUG("Creating new source %s", (source_name == NULL ? "(null)" : source_name));
-    new_source = g_new0 (gnc_quote_source, 1);
+    new_source = new gnc_quote_source;
```

Comment:
This change from `g_new0` to `new` caught my attention. You are in the process of converting a C-struct into a C++ one. There are plenty of places in our C code that are still accessing this as a C-struct  (like dialog-commodity.c, gnc-tree-model-commodity.c, ...). I don't know if this is a safe practice. I mean I don't know if the memory layout is guaranteed to be the same between a C and a C++ struct. So just a heads up from my side.

Reply:
And that aside, as a minor nitpick - in the commit history this would better fit with your next commit (where you also add a constructor to the struct) than in this commit (which is about replacing `const char *` with `std::string` in the definition.

Reply:
> I mean I don't know if the memory layout is guaranteed to be the same between a C and a C++ struct. So just a heads up from my side.

fixed by making struct members private, and using getters/setters.